src/ServiceControl.RavenDB/EmbeddedDatabase.cs

```diff
@@ -162,23 +178,87 @@ public async Task<IDocumentStore> Connect(CancellationToken cancellationToken)
 
         public async Task DeleteDatabase(string dbName)
         {
-            using var store = await EmbeddedServer.Instance.GetDocumentStoreAsync(new DatabaseOptions(dbName) { SkipCreatingDatabase = true });
+            using var store = await EmbeddedServer.Instance.GetDocumentStoreAsync(new DatabaseOptions(dbName)
+            {
+                SkipCreatingDatabase = true
+            });
             await store.Maintenance.Server.SendAsync(new DeleteDatabasesOperation(dbName, true));
         }
 
+        public async Task Stop(CancellationToken cancellationToken)
+        {
+            Logger.Debug("Stopping RavenDB server");
+            EmbeddedServer.Instance.ServerProcessExited -= OnServerProcessExited;
+
+            await shutdownTokenSource.CancelAsync();
+
+            // EmbeddedServer does not have have an async Stop method, the Dispose operation blocks and waits
+            // until its GracefulShutdownTimeout is reached and then does a Process.Kill
+            //
+            // This logic gets the child process ID uses a Task.Delay with infinite.
```

Comment:
This line makes no sense to me?

Reply:
@andreasohlund please review latest commit. I think this is an improvement.